Python tests are failing (and causing other agent tests to fail)

APM integration tests have been failing since November 9: https://apm-ci.elastic.co/job/apm-integration-tests/job/master/

In many of the runs the tests that fail are `test_flask_error` and `test_django_error`. In some others, other agents' equivalent tests are failing too.

Running the tests locally I observed that the document count on the `apm-8.0.0-error-*` indices keeps growing even after the tests are done, and they appear to all be coming from the Python agent. I think the failures in the non-Python agent tests are due to runaway exceptions in the Python service(s).

If I explicitly select Python agent 5.9.0 with `--python-agent-package='elastic-apm==5.9.0'`, then the tests pass.

Steps to reproduce:
 1. clone https://github.com/elastic/apm-integration-testing
 2. run `./scripts/compose.py start 8.0.0 --build-parallel  --no-apm-server-dashboards   --no-apm-server-self-instrument  --with-agent-python-django --with-agent-python-flask  --force-build   --no-xpack-secure   --apm-server-enable-tls --no-verify-server-cert    --apm-server-secret-token=SuPeRsEcReT   --apm-server-url=https://apm-server:8200   --apm-log-level=debug`
 3. `make venv`, then `. ./venv/bin/activate` and finally `pytest -v -s -m version tests/agent/test_python.py`

```
tests/agent/test_python.py::test_req_flask PASSED                   
tests/agent/test_python.py::test_flask_error FAILED                 
tests/agent/test_python.py::test_concurrent_req_flask PASSED        
tests/agent/test_python.py::test_req_flask_agent_config SKIPPED     
tests/agent/test_python.py::test_concurrent_req_flask_foobar PASSED
tests/agent/test_python.py::test_req_django PASSED 
tests/agent/test_python.py::test_django_error FAILED        
tests/agent/test_python.py::test_concurrent_req_django PASSED  
tests/agent/test_python.py::test_concurrent_req_django_foobar PASSED
```

```
django = <tests.fixtures.agents.Agent object at 0x7f2a5828d610>
    @pytest.mark.version
    @pytest.mark.django
    def test_django_error(django):
>       utils.check_agent_error(django.oof, django.apm_server.elasticsearch)
tests/agent/test_python.py:66: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/utils.py:23: in check_agent_error
    check_elasticsearch_count(elasticsearch, ct, processor='error')
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
elasticsearch = <tests.fixtures.es.es.<locals>.Elasticsearch object at 0x7f2a80eec9d0>
expected = 1, processor = 'error'
query = {'query': {'term': {'processor.name': 'error'}}}
    def check_elasticsearch_count(elasticsearch,
                                  expected,
                                  processor='transaction',
                                  query=None):
        if query is None:
            query = {'query': {'term': {'processor.name': processor}}}
    
        actual = 0
        retries = 0
        max_retries = 3
        while actual != expected and retries < max_retries:
            try:
                actual = elasticsearch.count(query)
                retries += 1
                time.sleep(10)
            except TimeoutError:
                retries += 1
                actual = -1
    
        assert actual == expected, "Expected {}, queried {}".format(
>           expected, actual)
E       AssertionError: Expected 1, queried 3621
tests/utils.py:62: AssertionError
```

Running these steps again with `--python-agent-package='elastic-apm==5.9.0'` in the `compose.py` invocation completes successfully.

--- a/docker/python/flask/app.py
+++ b/docker/python/flask/app.py
@@ -3,7 +3,6 @@
 import elasticapm
 from flask import Flask
 from elasticapm.contrib.flask import ElasticAPM
-from elasticapm.handlers.logging import LoggingHandler
 import logging
 import os
 
@@ -72,6 +71,6 @@ if __name__ == '__main__':
     app.run(host='0.0.0.0', port=int(os.environ['FLASK_PORT']))
 
     # Create a logging handler and attach it.
-    handler = LoggingHandler(client=apm.client)
+    handler = logging.StreamHandler()
     handler.setLevel(logging.INFO)
     app.logger.addHandler(handler)

--- a/docker/python/flask/app.py
+++ b/docker/python/flask/app.py
@@ -23,7 +23,7 @@ app.config['ELASTIC_APM'] = {
     'TRANSACTIONS_IGNORE_PATTERNS': ['.*healthcheck'],
     'LOG_LEVEL': 'warning',
 }
-apm = ElasticAPM(app, logging=True)
+apm = ElasticAPM(app, logging=False)
 
 
 @app.route('/')

--- a/tests/agent/test_python.py
+++ b/tests/agent/test_python.py
@@ -13,8 +13,7 @@ def test_req_flask(flask):
 @pytest.mark.version
 @pytest.mark.flask
 def test_flask_error(flask):
-    # one from exception handler, one from logging handler
-    utils.check_agent_error(flask.oof, flask.apm_server.elasticsearch, ct=2)
+    utils.check_agent_error(flask.oof, flask.apm_server.elasticsearch, ct=1)
 
 
 @pytest.mark.version

